packages/collector/test/tracing/misc/typescript/ts_esm/ts_esm_test.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * (c) Copyright IBM Corp. 2024
+ */
+
+'use strict';
+
+const expect = require('chai').expect;
+const path = require('path');
+const { execSync } = require('child_process');
+const config = require('@instana/core/test/config');
+const ProcessControls = require('../../../../test_util/ProcessControls');
+const globalAgent = require('../../../../globalAgent');
+const testUtils = require('@instana/core/test/test_util');
+
+describe('Typescript TS->ESM', function () {
+  this.timeout(config.getTestTimeout() * 5);
+
+  describe('[CASE 1]', () => {
+    globalAgent.setUpCleanUpHooks();
+    const agentControls = globalAgent.instance;
+
+    let controls;
+
+    beforeEach(async () => {
+      await agentControls.clearReceivedTraceData();
+    });
+
+    before(async () => {
+      execSync('rm -rf ./dist', { cwd: __dirname, stdio: 'inherit' });
+      execSync('rm -rf ./node_modules', { cwd: __dirname, stdio: 'inherit' });
+      execSync('npm install --verbose', { cwd: __dirname, stdio: 'inherit' });
+      execSync('npm run build', { cwd: __dirname, stdio: 'inherit' });
+      execSync('cp ./package.json.tmp ./dist/package.json', { cwd: __dirname, stdio: 'inherit' });
+
+      controls = new ProcessControls({
+        appPath: path.join(__dirname, 'dist', 'app_1.js'),
+        useGlobalAgent: true,
+        execArgv: ['--import', path.join(__dirname, 'node_modules', '@instana', 'collector', 'esm-register.mjs')]
```

Comment:
What will be happen if we run this in lower nodejs versions? The command with--import work only for 18.19 and above?

Reply:
Yes 👍 Will add the conditional check thanks

Reply:
Changed